Multi sensitive volume clusters

ybedfer opened the pull request:
### Briefly, what does this PR introduce?
Implement realistic pitch into Multi-SensitiveVolume-MPGD, extending PR#https://github.com/eic/epic/pull/1014.

### What kind of change does this PR introduce?
- [* ] New feature, see PR#2264 ("https://github.com/eic/EICrecon/pull/2264").

### Please check if this PR fulfills the following:
- [ *] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No

### Does this PR change default behavior?
Eventually yes.

Commit:
MPGD Barrel: Add "5SensVolumes" versions of XML. Modified Detector Constructors.
- "5SensVolumes" are designed for a 2D-strip readout, via "MPGDTrackerDigi".
- Modified constructors provide for both standard and "5SensVolumes".
- OuterBarrel constructor:
+ Simplified: providing now for only one <module> and one <layer> tag.
+ Sensisitive DetElement: in line with other constructors ("volSurfaceList",...).

Commit:
Merge branch 'main' into Multi-SensitiveVolume-MPGD

Commit:
MPGD Barrel: Simplification. "5SensVolumes" renamed "2DStrip". Add dedicated configuration.
- Simplication: Standard XML (almost-)free of any reference to strips.
- Dedicated "craterlake_tracking_only_2DStrip" implementing "2DStrip" XMLs.

Commit:
Merge branch 'main' into Multi-SensitiveVolume-MPGD

Commit:
MPGD Barrel: Backward compatibility via cmake option "-DMPGD_INSTALL_PIXEL".
- 2 flavours of Barrel MPGDs implemented in 2 distinct XMLs: "_pixel"/"_2DStrip".
- "-DMPGD_INSTALL_PIXEL" allows to install "_pixel" instead of default "_2DStrip".

Commit:
MPGD Barrel: Install "_pixel" flavors of XMLs by default. Erase pre-existing files in "install" directory.
- "_pixel" by default: so that the PR checks run successfully even when the new version of "MPGDTrackerDigi" (see "https://github.com/eic/EICrecon/pull/2177") is not in place.
- Pre-existing files explicitly erased: since the cmake command "install" does not overwrite existing files.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

ybedfer committed:
Merge branch 'main' into Multi-SensitiveVolume-MPGD

ybedfer committed:
Merge branch 'main' into Multi-SensitiveVolume-MPGD

Commit:
MPGD: Cancel switching _pixel/_2DStrip from cmake. Add a special _2DStrip configurations.

ybedfer committed:
MPGD: 2DStrip,v.2 (realistic pitch and correct apperture) StripID from XML.
- 2DStrip:
+ CyMBaL: v2.1 is the 5-SUBVOLUME version, v2.2 is clusterisation compatible.
+ Outer: same thing w/ v1.1 and v1.2.
- StripID: Merge in #767a3c7a0 of branch Multi-SensitiveVolume-MPGD.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

ybedfer committed:
Barrel MPGDs: Merge with comit #896b85e80.

ybedfer committed:
Merge branch 'main' into Multi-SensitiveVolume-Clusters

ybedfer committed:
CyMBaL detector builder: cosmetics (nicer layout).

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

veprbl committed:
Merge branch 'main' into Multi-SensitiveVolume-Clusters

veprbl commented on compact/tracking/mpgd_outerbarrel_2DStrip.xml:
```diff
@@ -181,9 +184,9 @@
-        <segmentation name="BotRad"  type="CartesianGridUV" key_value="3" grid_size_u="0.150*mm*sqrt(12)"    grid_size_v="0.150*mm*sqrt(12)"    grid_angle="pi*rad/4" />
+        <segmentation name="UStrips" type="CartesianGridUV" key_value="1" grid_size_u="MPGDOuterBarrelPitch" grid_size_v="MPGDOuterStrip_range" grid_angle="pi*rad/4" offset_u="MPGDOuterBarrelPitch/2" />
+        <segmentation name="VStrips" type="CartesianGridUV" key_value="2" grid_size_u="MPGDOuterStrip_range" grid_size_v="MPGDOuterBarrelPitch" grid_angle="pi*rad/4" offset_v="MPGDOuterBarrelPitch/2" />
+        <segmentation name="BotRad"  type="CartesianGridUV" key_value="3" grid_size_u="0.150*mm*sqrt(12)"    grid_size_v="MPGDOuterBarrelPitch" grid_angle="pi*rad/4" />
```
Why only v is updated only for BotRad and not TopRad?

ybedfer replied:
Hello,
It's a mistake.
But a mistake that has no consequence. Or so I think. The Radiators are only used at simulation time (`npsim`) and the CellID that is then computed _via_ segmentation and assigned to the SimHits is later on disregarded. Only the EDep of the SimHits. matters.
Yet, I think that the best is to have a pixel segmentation in those Radiators. I have corrected the XML file and committed.
Thank you for your careful review.

veprbl replied:
I take it that pixel resolution is better than resolution of true strip dimensions, so that does not hurt because you rely on coordinates. Why not make pixels match the dimensions of strips? Otherwise you might experience small effects due to pitch mismatch.

ybedfer replied:
Hello,

> Why not make pixels match the dimensions of strips? Otherwise you might experience small effects due to pitch mismatch.

I don't know whether it's a good idea.
On the one hand, it worsens what I would call the readability of the podio `TTree`:  the readability of  what I get when I  `TTree:Scan` it, that is. When the segmentation has a pitch equal to the resolution, I can find back in the CellIDs of my RecHits, pieces of the CellIDs of my original SimHits. That's this is a genuine advantage is questionable, I agree. So more so as we have the opposite situation when comparing SimHits to RawHits.
On the other hand, there's what you call "pitch mismatch". But I don't get what this might be. Can you be more precise?

veprbl replied:
All I'm saying is that if you sample at discrete pixel pitch coordinates and then essentially histogram those with strip pitch you will get "beat signal" modulations in the resulting distribution.

veprbl replied:
If this is not an issue, let's mark it resolved and we can move to merging.

ybedfer replied:
Hello,
I have added comments describing the alternative for what I call Helper Sensitive Volumes: either resolution or pitch.

veprbl replied:
Thanks for updating the comments. I want to check with you if they are misleading. You discuss helper volumes that may not necessarily have adequate segmentation, e.g.:

> - Two of the Sensitive Volumes, '(p|n)ThinGap", have a meaningful Segmentation, with a pitch corresponding to that of the actual detector.

Yet those correspond to BotRad and TopRad segmentation, and don't have the physical pitch.

ybedfer replied:
I have been fighting with the English of the comments. Not sure I have been successful...
The confusion may come from the fact that it's not clearly stated that to each of the 5 sensitive volumes corresponds a distinct Segmentation. Added to the fact  that Volumes and Segmentation entries do not have the same name. These names cannot be strictly the same (because of CyMBaL where one has to in addition distinguish between inner and outer sections). But they can be made more similar. I am preparing a new commit where I implement a better naming scheme.

veprbl replied:
Okay, maybe I got confused. Those are not BotRad and TopRad. Those are U/VStrips.

Maybe my confusion now is with then you say
> The other three Volumes are Helpers, used only temporarily in the production of MC data.

Not sure if "Helpers" is a good word. They define a pixel binning that is used by the simulation.

ybedfer replied:
Hello,
I committed a new attempt.
Where I still have the word "Helpers" (sorry: I had not read your message before committing).
By "Helpers" I mean something that comes into play only temporarily, for the time MC data are produced and digitized, and are then forgotten. In the processing of real data, they are even completely ignored.

veprbl replied:
I like latest change to the naming. Before you could still look at `key`, but this helps even more.

veprbl reviewed (dismissed):
LGTM

ybedfer committed:
MPGDOuterBarrel: Fix mistake in XML: now both Radiator subVolumes have pixel-like segmentation.
The actual segmentation of the Radiator subVolumes is unimportant (being disregarded at
digitization time). But a pixel-like segmentation gives to the simulated data a better outlook.
And now the setting is consistent and consistent w/ that of CyMBaL.

ybedfer committed:
Merge branch 'main' into Multi-SensitiveVolume-Clusters

ybedfer committed:
MPGD XML: Commenting the two alternatives for the Segmentation of Helper Sensitive Volumes.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

ybedfer committed:
MPGD XML: New attempt at improving the comments.

ybedfer committed:
Merge branch 'Multi-SensitiveVolume-Clusters' of github.com:eic/epic into Multi-SensitiveVolume-Clusters

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

veprbl reviewed (approved):
LGTM

veprbl committed:
Merge branch 'main' into Multi-SensitiveVolume-Clusters
